src/node.cc

```diff
@@ -3901,9 +3906,7 @@ Environment* CreateEnvironment(Isolate* isolate,
       HandleCleanup,
       nullptr);
 
-  if (v8_is_profiling) {
-    StartProfilerIdleNotifier(env);
-  }
+  StartProfilerIdleNotifier(env);
```

Comment:
Maybe you should check what StartProfilerIdleNotifier() does before making such assertions.